### PR TITLE
LongShot Command

### PR DESCRIPTION
**Pull request by shinysocks**
added LongShot command on operator X button press. This SequentialCommandGroup moves arm to 94°, Spins up shooter motors to 8k rpm, and fires a note.

### CODE REVIEW

**Comment by shinysocks** [comment]
needs testing..

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -58,10 +58,12 @@ public RobotContainer() {
 
     robotContext = new RobotContext(arm);
 
-    autoGenerator = Constants.PathPlanner.PATH_PLANNER_IS_ENABLED
-
-        ? new PathPlannerAutoGenerator(drive, arm, shooter, undertakerSubsystem, robotContext)
-        : new SimpleAutoGenerator(drive, shooter, undertakerSubsystem, robotContext);
+    autoGenerator = 
+      Constants.PathPlanner.PATH_PLANNER_IS_ENABLED
+      ?
```
Comment (pordonj) [inline src/main/java/frc/robot/RobotContainer.java:63]: if a ternary is this long i'd prefer you just do it the long way, lol

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -51,6 +51,9 @@ public class Arm {
         // PID Profile constants
         public static final double MAXIMUM_VELOCITY = 150; // degrees per second
         public static final double MAXIMUM_ACCELERATION = 750; // degrees per second squared
+
+    		public static final double LONG_SHOT_ANGLE = 94;
```
Comment (pordonj) [inline src/main/java/frc/robot/Constants.java:55]: There are tabs in this line, set up your IDE to not use tabs!

**Review by pordonj — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (pordonj) [reply src/main/java/frc/robot/RobotContainer.java:63]: Up to you though!

Comment (pordonj) [reply src/main/java/frc/robot/RobotContainer.java:63]: Hold on this isn't even for your changes. Don't do this in this PR please.

Comment (shinysocks) [reply src/main/java/frc/robot/RobotContainer.java:63]: i accidentally modified it in this PR without meaning to...(auto format) which is why I set it back

File: src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (shinysocks) [reply src/main/java/frc/robot/Constants.java:55]: accidental manual tab because my indentation is set to 2 and i just wanted it to get over there

Comment (shinysocks) [reply src/main/java/frc/robot/Constants.java:55]: i'll remove it

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (pordonj) [reply src/main/java/frc/robot/RobotContainer.java:63]: ohhh sorry

File: src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (pordonj) [reply src/main/java/frc/robot/Constants.java]: your editor should have some way to make pressing tab just put spaces idk

**Comment by shinysocks** [comment]
unclear if the tabs were removed..

File: src/main/java/frc/robot/shooter/commands/SpinUp.java [hunk 3]
```diff
@@ -9,17 +9,26 @@
...
+    this.subsystem = subsystem;
+    this.slowShoot = () -> false;
+    this.targetSpeed = targetSpeed;
     addRequirements(subsystem);
   }
 
   @Override
   public void initialize() {
     double scale = slowShoot.getAsBoolean() ? Constants.Shooter.AMP_SHOOT_SCALE : 1;
-    subsystem.setTargetShooterSpeed(Constants.Shooter.SHOOT_SPEED * scale);
```
Comment (pordonj) [inline src/main/java/frc/robot/shooter/commands/SpinUp.java:22]: This is why i think you need to switch to separate targets, if you want the close shot to remain different. Right now the long shot would shoot at 8000 and 7200.

https://github.com/frc1675/frc1675-2024/blob/main/src/main/java/frc/robot/shooter/ShooterSubsystem.java#L63

**Review by pordonj — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/shooter/commands/SpinUp.java (hunk 3 above)
Comment (pordonj) [reply src/main/java/frc/robot/shooter/commands/SpinUp.java:22]: This comment was supposed to be on the new version of this

**Comment by pordonj** [comment]
the tabs are gone

**Comment by pordonj** [comment]
you'll need to manually resolve conflicts because of the other PR (just take whatever main has)

File: src/main/java/frc/robot/cmdGroup/SpinUpAndShoot.java [hunk 4]
```diff
@@ -10,7 +10,7 @@
 public class SpinUpAndShoot extends SequentialCommandGroup {
   public SpinUpAndShoot(ShooterSubsystem shooter, BooleanSupplier slowShoot) {
     addCommands(
-        new SpinUp(shooter, slowShoot),
+        new SpinUp(shooter, Constants.Shooter.SHOOT_SPEED, slowShoot),
```
Comment (pordonj) [inline src/main/java/frc/robot/cmdGroup/SpinUpAndShoot.java:13]: "resolving the merge conflict" means making this change in SUAS in the new place, too.

File: src/main/java/frc/robot/shooter/commands/SpinUp.java (hunk 3 above)
Comment (shinysocks) [reply src/main/java/frc/robot/shooter/commands/SpinUp.java:22]: I'm confused? are you referring to the wheels running at different speeds?

**Comment by shinysocks** [comment]
could you look at this?

**Review by shinysocks — COMMENTED** [review]
it says I have changes to review? I'm not entirely sure what this means

**Review by pordonj — APPROVED** [review]

**Comment by pordonj** [comment]
Do not merge until tested - if the subwoofer shot works and the long shot doesnt by end of wednesday this can be merged and worked on at competition

**Comment by shinysocks** [comment]
> Do not merge until tested - if the subwoofer shot works and the long shot doesnt by end of wednesday this can be merged and worked on at competition

Do we like push to main at competition? how does that work?

**Pull request merged**